src/net_processing.cpp

```diff
@@ -818,14 +830,14 @@ void PeerLogicValidation::InitializeNode(CNode *pnode) {
 
 void PeerLogicValidation::ReattemptInitialBroadcast(CScheduler& scheduler) const
 {
-    std::set<uint256> unbroadcast_txids = m_mempool.GetUnbroadcastTxs();
+    std::map<uint256, uint256> unbroadcast_txids = m_mempool.GetUnbroadcastTxs();
 
-    for (const uint256& txid : unbroadcast_txids) {
+    for (const std::pair<uint256, uint256>& elem : unbroadcast_txids) {
```

Comment:
from a quick look, seems like using `const` and reference declarator together on this `std::pair<uint256,uint256>` type is causing the build warnings. I don't yet understand why. 

leaving `&` but removing the `const` throws an error: 
```
net_processing.cpp:835:39: error: non-const lvalue reference to type 'pair<uint256, [...]>' cannot bind to a value of unrelated type 'pair<const uint256, [...]>'
    for (std::pair<uint256, uint256>& elem : unbroadcast_txids) {
```

but removing the reference declarator allows it to build with no errors / warnings. with or without the `const`.

Reply:
The reason is that the element type of `unbroadcast_txids` is `std::pair<const uint256, uint256>` which can't be converted to `std::pair<uint256, uint256>`. If you want a const reference to such a thing, it will be copied, and you'll get a reference to the copy.

The correct solution is either (`const`) `std::pair<const uint256, uint256>&`, or just `const auto&`.

Reply:
Fixed with `const auto&`.